iotit.go

```diff
@@ -1,204 +1,181 @@
-			Version = "no version"
-		}
-
-		fmt.Println(progName, Version)
-	}
-
-	h := func() {
-		fmt.Println(helpInfo)
-	}
-
-	i := func() {
-		logrus.Debug("Checking ", installPath, progName)
-		if _, err := os.Stat(installPath + progName); os.IsNotExist(err) {
-			p, err := os.Executable()
-			if err != nil {
-				logrus.Fatal("[-] ", err)
-			}
-
-			fmt.Println("[+] You may need to enter your user password")
-			_, eut, err := sudo.Exec(sudo.InputMaskedPassword, nil, "cp", p, installPath+progName)
-			fmt.Println("[+] Copying", p, installPath+progName)
-			if err != nil {
-				fmt.Println("[-] Error: ", string(eut))
-				return
-			}
-
-			fmt.Println("[+] Done")
-
-			return
-		}
-
-		fmt.Println("[+] Software is already installed")
-	}
-
-	u := func() {
-		logrus.Debug("Checking ", installPath, progName)
-		if _, err := os.Stat(installPath + progName); os.IsNotExist(err) {
-			fmt.Println("[+] Software is not installed")
-			return
-		}
-
-		fmt.Println("[+] You may need to enter your user password")
-		_, eut, err := sudo.Exec(sudo.InputMaskedPassword, nil, "rm", installPath+progName)
-		fmt.Println("[+] Removing", installPath+progName)
-		if err != nil {
-			fmt.Println("[-] Error: ", string(eut))
-			return
-		}
-
-		fmt.Println("[+] Done")
+	app := cli.NewApp()
+	app.Version = Version
+	app.Name = progName
+	app.Usage = "Flashing Tool for iot devices used by Isaax Cloud"
+
+	app.Action = func(c *cli.Context) error {
+		// TODO: launch gui by default
+		device.Init(c.Args().Get(0))
+		return nil
 	}
 
-	upd := func() {
-		if _, err := os.Stat(installPath + progName); os.IsNotExist(err) {
-			fmt.Println("[-] Software is not installed, please install it globally first: `" + progName + " gl`")
-			return
-		}
-
-		fmt.Println("[+] Current os: ", runtime.GOOS, runtime.GOARCH)
-
-		dir, err := repo.DownloadNewVersion(progName, Version, help.GetTempDir())
-
-		if err != nil {
-			fmt.Println("[-] Error:", err)
-			return
-		}
-
-		if dir == "" {
-			fmt.Println("[+] ", progName, " is up to date")
-		} else {
-
-			fmt.Println("[+] You may need to enter your user password")
-
-			if _, eut, err := sudo.Exec(sudo.InputMaskedPassword, nil, "mv", dir, installPath+progName); err != nil {
-				fmt.Println("[-] Error:", eut)
-				return
-			}
-			fmt.Println("[+]", progName, " is updated")
-		}
-
-		if vbox.CheckUpdate() {
-			if dialogs.YesNoDialog("Would you like to update vbox?") {
-				vbox.Update()
-			}
-		}
+	app.Commands = []cli.Command{
+		{
+			Name:    "flash",
+			Aliases: []string{"f"},
+			Usage:   "Flash image to the device",
+			Action: func(c *cli.Context) error {
+				device.Init(c.Args().Get(0))
+				return nil
+			},
+		},
+		{
+			Name:    "install",
+			Aliases: []string{"i"},
+			Usage:   "Install to global app environment",
+			Action: func(c *cli.Context) error {
+				log.Debug("Checking ", installPath, progName)
+				if _, err := os.Stat(installPath + progName); os.IsNotExist(err) {
+					p, err := os.Executable()
```

Comment:
__[gometalinter]__
* [unconvert] Executable not declared by package os
* [interfacer] Executable not declared by package os
* [staticcheck] Executable not declared by package os
* [gosimple] Executable not declared by package os

<!-- Comment by [SideCI](https://sideci.com) -->